### ISSUE TITLE
 <dd4t:componentpresentations  view="XXX" /> not working

### ISSUE THREAD

**Issue by anbusampath** [dd4t/dd4t-2-java#48]
We have component template with metadata field "view". BY using dd4t Tag components are not getting resolved.

Rootcause : 
In BaseComponentPresentationsTag view attribute value converted to lower case.
```java
 public String getView () {
        return view;
    }

    public void setView (final String view) {
        this.view = view.toLowerCase(Locale.getDefault());
    }
```

In RenderUtils we are comparing whatever value we received from metadata without converting to lowercase.
```java
 String view = getViewName(cp);

                if (match(componentSchema.toLowerCase(), schemaName) &&
                        match(componentRootElement, rootElementName) &&
                        match(view, viewName) && isComponentPresentationInRegion(region, regionIsSet, cp)) {
                    componentPresentationsToRender.add(cp);
                }
```
.
.
//Get View name not doing the lowercase conversion.
```java
 public static String getViewName (final ComponentPresentation cp) {
        ComponentTemplate componentTemplate = cp.getComponentTemplate();
        String viewNameKey = DataBindFactory.findComponentTemplateViewName(componentTemplate);

        if (StringUtils.isNotEmpty(viewNameKey)) {
            return viewNameKey;
        }

        return componentTemplate.getTitle().toLowerCase();
    }
```

To fix the issue we need to convert viewNameKey to LowerCase as well.

```Java
 if (StringUtils.isNotEmpty(viewNameKey)) {
            return viewNameKey.toLowerCase();
        }
```

**Comment by RaimondKempees** [comment]
Hi @anbusampath,

I think the best thing to do is to remove the toLowerCase altogether. That way the set view name in the CP metadata becomes leading, without any surprises. What do you think?

**Comment by anbusampath** [comment]
Yeah,  I think its good option to remove lowerCase convention altogether.
So we can use same name in metadata.

On 11 April 2016 at 13:26, RaimondKempees <notifications@github.com> wrote:

> Hi @anbusampath <https://github.com/anbusampath>,
>
> I think the best thing to do is to remove the toLowerCase altogether. That
> way the set view name in the CP metadata becomes leading, without any
> surprises. What do you think?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/dd4t/dd4t-2-java/issues/48#issuecomment-208212500>
>



-- 
--Anbu

**Comment by RaimondKempees** [comment]
Yeah. Fixed on Dev. Available in dd4t-2.0.5
